src/test/scala/is/hail/methods/ExprSuite.scala

```diff
@@ -585,6 +588,56 @@ class ExprSuite extends SparkSuite {
     // FIXME: parser should accept minimum Long/Int literals
     // assert(eval[Long](Long.MinValue.toString+"L").contains(Long.MinValue))
     // assert(eval[Long](Long.MinValue.toString+"l").contains(Long.MinValue))
+
+    {
+      val x = eval[Map[String,IndexedSeq[Int]]]("[1,2,3,4,5].groupBy(k => if (k % 2 == 0) \"even\" else \"odd\")")
+      assert(x.isDefined)
+      x.get.keySet should contain theSameElementsAs Seq("even", "odd")
```

Comment:
I agree whole-hearted, writing tests (or any computer algorithm) as natural English is shit. I just want intelligent matchers that compose and produce useful error messages. I'm disappointed with the response of the scalatest developer to https://github.com/scalatest/scalatest/issues/471, which AFAICT, reaffirms the commitment to english-language tests, not to tests that explain what went wrong.

I suspect we could build a matcher library to do what I want below. Hopefully that would also compose well with the quick check library.